src/Altinn.App.Core/Features/ExternalApi/ExternalApiFactory.cs

```diff
@@ -0,0 +1,51 @@
+using Microsoft.Extensions.Logging;
+
+namespace Altinn.App.Core.Features.ExternalApi;
+
+/// <summary>
+/// Interface of factory class for resolving <see cref="IExternalApiClient"/> implementations
+/// </summary>
+public interface IExternalApiFactory
+{
+    /// <summary>
+    /// Finds the implementation of <see cref="IExternalApiClient"/> based on the external api id
+    /// </summary>
+    IExternalApiClient GetExternalApiClient(string externalApiId);
+}
+
+/// <summary>
+/// Factory class for resolving <see cref="IExternalApiClient"/> implementations
+/// </summary>
+public class ExternalApiFactory : IExternalApiFactory
+{
+    private readonly ILogger<ExternalApiFactory> _logger;
+    private IEnumerable<IExternalApiClient> _externalApiClients { get; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ExternalApiFactory"/> class.
+    /// </summary>
+    /// <param name="logger"></param>
+    /// <param name="externalApiClients"></param>
+    public ExternalApiFactory(ILogger<ExternalApiFactory> logger, IEnumerable<IExternalApiClient> externalApiClients)
+    {
+        _logger = logger;
+        _externalApiClients = externalApiClients;
+    }
+
+    /// <inheritdoc/>
+    /// <param name="externalApiId"></param>
+    public IExternalApiClient GetExternalApiClient(string externalApiId)
+    {
+        var client = _externalApiClients.FirstOrDefault(e =>
+            e.Id.Equals(externalApiId, StringComparison.OrdinalIgnoreCase)
+        );
+
+        if (client == null)
+        {
+            _logger.LogWarning("No external client found for external API with id {ExternalApiId}", externalApiId);
+            throw new KeyNotFoundException($"No external api client found for external API with id {externalApiId}");
```

Comment:
Should we return `null` here instead? Returning `IExternalApiClient?` sSeems simpler than throwing an exception, it's not documented in xmldocs either

Reply:
Hm, ser ut som det er litt forskjellige meninger på nettet. Blant annet at man eventuelt kan returnere 204. Men om man ikke finner det eksterne api-et er det jo en feil, så syns kanskje ikke 200-serien er helt riktig? 🤔 og ved å returnere `200(null)` så lurer jeg på hvordan brukeren skal vite at et eksternt api med den satte id-en ikke eksisterer og ikke at resultatet av å hente data fra det eksterne api-et er `null`?

Reply:
Whoops engelsk 😅 
With regards to the documentation, do you have any suggestions on how to document it? It's not an exception, so I don't think that would be right. Saw something about a `<response>` tag, but that isn't documentet in the [dotnet documentation](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/) and doesn't work with intellisense.

Reply:
I was only referring to the `IExternalApiFactory.GetExternalApiClient` method, not the HTTP status code. I see that it is 404 now, but isn't the issue that the input is wrong? Should it be  400? Reasons for my `IExternalApiFactory` suggestion:

* Exceptions tend to be used for exceptional cases (unexpected conditions), is someone trying to refer to a non-existing external API exceptional?
* I think `IExternalApiClient? GetExternalApiClient(string externalApiId);` more clearly communicates the behavior, it let's the caller know just from the method signature that it either returns what was asked for or `null`
* Exceptions are poor for performance

XML doc for exception is `<exception cref="KeyNotFoundException">some message</exception>`

Reply:
Ahh, I see I was looking at the controller, not the factory. My bad! I think it makes sense to return null in the factory, but still return 404 in the controller. Did some changes according to that [in the last commit](https://github.com/Altinn/app-lib-dotnet/pull/700/commits/69464a2b765dfa75b47dcf718890159ed2bd7b06) if you'd like to check it out 😊